evalml/model_understanding/graphs.py

```diff
@@ -608,12 +617,16 @@ def partial_dependence(pipeline, X, features, percentiles=(0.05, 0.95), grid_res
     """
 
     X = infer_feature_types(X)
-    # Dynamically set the grid resolution to the maximum number of categories
-    # in the categorical variables if there are more categories than resolution cells
+    # Dynamically set the grid resolution to the maximum number of values
+    # in the categorical/datetime variables if there are more categories/datetime values than resolution cells
     X_cats = X.select("categorical")
     if X_cats.shape[1] != 0:
         max_num_cats = max(X_cats.describe().loc["nunique"])
         grid_resolution = max([max_num_cats + 1, grid_resolution])
+    X_dt = X.select("datetime")
```

Comment:
I think this falls into the same problem as #2116. I think we should only update the grid resolution if the requested feature is `datetime`